libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificatePinnerFactory.java

```diff
@@ -36,11 +37,37 @@ CertificatePinner provideCertificatePinnerFor(Environment environment) {
     return CERTIFICATES_PINS.get(environment);
   }
 
-  private void addCertificatesPins(Map<String, List<String>> pins, CertificatePinner.Builder builder) {
+  private void addCertificatesPins(Map<String, List<String>> pins, CertificatePinner.Builder builder,
+                                   CertificateBlacklist certificateBlacklist) {
+    pins = removeBlacklistedPins(pins, certificateBlacklist);
```

Comment:
According to the name of this method it should only add pins to the blacklist, now it's actually also does cleanup, hence would it make sense to decouple cleanup logic and call removeBlacklistedPins on line 30 just right before addCertificatesPins is called?